e2etests/test/twin_e2e_tests.js

```diff
@@ -190,27 +190,38 @@ delete nullMergeResult.tweedle;
 
     var sendsAndReceivesDesiredProperties = function(done) {
       assertObjectIsEmpty(deviceTwin.properties.desired);
+      deviceTwin.on('properties.desired', function(props) {
+        if (props.$version === 1) {
+          // ignore
+          assert(true);
+        } else if (props.$version === 2) {
+          assertObjectsAreEqual(newProps, deviceTwin.properties.desired);
+          done();
+        } else {
+          done(new Error('incorrect property version received - ' + props.$version));
+        }
+      });
 
       serviceTwin.update( { properties : { desired : newProps } }, function(err) {
         if (err) return done(err);
```

Comment:
scary re-ordering of test.   I'm not saying this change is bad, but why did you need to change this?  (I'm wondering if this is related to the unsubscribe)

Reply:
the reason is that if you fire the update first, and then have the device subscribe, then the update to desired properties might happen before your subscription, which means that you'll never get an event for that - subscribing first and then calling serviceTwin.update makes sure we're subscribed first (well - sortof. since we don't wait on the SUBACK because deviceTwin.on is synchronous.... still. better this way i think.

---
In reply to: [170712640](https://github.com/Azure/azure-iot-sdk-node/pull/236#discussion_r170712640) [](ancestors = 170712640)